tests/testsuite/patch.rs

```diff
@@ -787,6 +787,154 @@ fn add_ignored_patch() {
         .run();
 }
 
+#[cargo_test]
+fn add_patch_with_features() {
+    Package::new("bar", "0.1.0").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+                authors = []
+
+                [dependencies]
+                bar = "0.1.0"
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .file("bar/Cargo.toml", &basic_manifest("bar", "0.1.0"))
+        .file("bar/src/lib.rs", r#""#)
+        .build();
+
+    p.cargo("build")
+        .with_stderr(
+            "\
+[UPDATING] `[ROOT][..]` index
+[DOWNLOADING] crates ...
+[DOWNLOADED] bar v0.1.0 [..]
+[COMPILING] bar v0.1.0
+[COMPILING] foo v0.0.1 ([CWD])
+[FINISHED] dev [unoptimized + debuginfo] target(s) in [..]
+",
+        )
+        .run();
+    p.cargo("build").with_stderr("[FINISHED] [..]").run();
```

Comment:
I know this is nit-picky, but would you mind removing these builds without the patch?  I think it would help keep the test smaller and more straight to the point.  Also, each build adds a few seconds. Individually that doesn't matter, but multiplied times hundreds of tests, it can add up, so it is good to try to keep each test as streamlined as it can.

Reply:
Make sense. Fixed. Thanks for your review!